0013-libxl-stubdom-add-kernel-ramdisk-options.patch

```diff
@@ -0,0 +1,106 @@
+From: Dmitry Fedorov <d.fedorov@tabit.pro>
+Subject: [PATCH] libxl: add support for emulator' kernel and ramdisk path
+
+Add ability to configure emulator kernel and ramdisk path.
+
+Signed-off-by: Dmitry Fedorov <d.fedorov@tabit.pro>
+
+diff -bpru libvirt-6.6.0/docs/schemas/domaincommon.rng libvirt-6.6.0.new/docs/schemas/domaincommon.rng
+--- libvirt-6.6.0/docs/schemas/domaincommon.rng	2020-12-23 19:05:01.937312919 +0300
++++ libvirt-6.6.0.new/docs/schemas/domaincommon.rng	2020-12-23 19:01:16.789267904 +0300
+@@ -3502,6 +3502,16 @@
+               <ref name="memoryKB"/>
+             </attribute>
+           </optional>
++          <optional>
++            <attribute name="kernel">
++              <ref name="absFilePath"/>
++            </attribute>
++          </optional>
++          <optional>
++            <attribute name="ramdisk">
++              <ref name="absFilePath"/>
++            </attribute>
++          </optional>
+           <ref name="absFilePath"/>
+         </group>
+         <group>
+@@ -3517,6 +3527,16 @@
+               <ref name="memoryKB"/>
+             </attribute>
+           </optional>
++          <optional>
++            <attribute name="kernel">
++              <ref name="absFilePath"/>
++            </attribute>
++          </optional>
++          <optional>
++            <attribute name="ramdisk">
++              <ref name="absFilePath"/>
++            </attribute>
++          </optional>
+           <empty/>
+         </group>
+       </choice>
+diff -bpru libvirt-6.6.0/src/conf/domain_conf.c libvirt-6.6.0.new/src/conf/domain_conf.c
+--- libvirt-6.6.0/src/conf/domain_conf.c	2020-12-23 19:05:01.941312938 +0300
++++ libvirt-6.6.0.new/src/conf/domain_conf.c	2020-12-23 18:59:44.848841163 +0300
+@@ -21711,6 +21711,8 @@ virDomainDefParseXML(xmlDocPtr xml,
+         }
+     }
+     def->emulator_cmdline = virXPathString("string(./devices/emulator/@cmdline)", ctxt);
++    def->emulator_kernel = virXPathString("string(./devices/emulator/@kernel)", ctxt);
++    def->emulator_ramdisk = virXPathString("string(./devices/emulator/@ramdisk)", ctxt);
```

Comment:
I think you need to adjust `virDomainDefFree` to release this memory. Which made me realize, `0003-libxl-Stubdom-emulator-type.patch` is buggy in this regard...